File: src/test/java/lrusso96/simplebiblio/FeedbooksTest.java
package lrusso96.simplebiblio;

import lrusso96.simplebiblio.core.Ebook;
import lrusso96.simplebiblio.core.providers.feedbooks.Feedbooks;
import lrusso96.simplebiblio.core.providers.feedbooks.FeedbooksBuilder;
import lrusso96.simplebiblio.core.providers.feedbooks.Language;
import lrusso96.simplebiblio.exceptions.BiblioException;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.*;

public class FeedbooksTest {
    @Test
    public void searchTest() throws BiblioException {
        Feedbooks feedbooks = new FeedbooksBuilder().build();
        List<Ebook> ret = feedbooks.search("Carroll");
        simpleTest(ret);
    }

    @Test
    public void recentTest() throws BiblioException {
        Feedbooks feedbooks = new FeedbooksBuilder().build();
        List<Ebook> ret = feedbooks.getRecent();
        simpleTest(ret);
    }

    private void simpleTest(List<Ebook> ret) {
        assertNotEquals(0, ret.size());
        Ebook book = ret.get(0);
        assertNotNull(book.getAuthor());
        assertNotEquals(0, book.getId());
        assertNotNull(book.getTitle());
        assertNotNull(book.getDownload());
        assertNotEquals(0, book.getDownload().getExtension().length());
    }

    @Test
    public void customLanguageTest() throws BiblioException {
        Feedbooks feedbooks = new FeedbooksBuilder().addLanguage(Language.ITALIAN).build();
        List<Ebook> ret = feedbooks.search("Dante Alighieri");
        assertNotEquals(0, ret.size());
        Ebook book = ret.get(0);
        assertNotNull(book.getAuthor());
        assertEquals("it", book.getLanguage());
    }
}

File: src/test/java/lrusso96/simplebiblio/FeedbooksTest.kt
package lrusso96.simplebiblio

import kotlinx.coroutines.runBlocking
import lrusso96.simplebiblio.core.Ebook
import lrusso96.simplebiblio.core.providers.feedbooks.Feedbooks
import lrusso96.simplebiblio.core.providers.feedbooks.Language
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class FeedbooksTest {
    @Test
    fun searchTest() {
        val feedbooks = Feedbooks.Builder().build()
        val ret = runBlocking { feedbooks.search("Carroll") }
        simpleTest(ret)
    }

    @Test
    fun recentTest() {
        val feedbooks = Feedbooks.Builder().build()
        val ret = runBlocking { feedbooks.getRecent() }
        simpleTest(ret)
    }

    private fun simpleTest(ret: List<Ebook>) {
        assertTrue(ret.isNotEmpty())
        val book = ret[0]
        assertNotNull(book.author)
        assertNotEquals(0, book.id)
        assertNotNull(book.title)
        assertTrue(book.getDownloads().isNotEmpty())
        assertTrue(book.getDownloads()[0].extension!!.isNotBlank())
    }

    @Test
    fun customLanguageTest() {
        val feedbooks = Feedbooks.Builder().addLanguage(Language.ITALIAN).build()
        val ret = runBlocking { feedbooks.search("Dante Alighieri") }
        assertTrue(ret.isNotEmpty())
        val book = ret[0]
        assertNotNull(book.author)
        assertEquals("it", book.language)
    }
}